### PR TITLE
port toranj test42 meshcop_joiner_discerner

### PR DESCRIPTION
**Pull request by priyankaCh22**

### CODE REVIEW

File: silk/tests/openthread/ot_test_meshcop_joiner_discerner.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        self.commissioner.commissioner_start()
+        self.wait_for_completion(self.device_list)
+
+        # Add `joiner1` and `joiner2` using joiner discerner.
+        self.commissioner.commissioner_add_joiner_with_discerner(DISCERNER1, D_LEN1, PSK1, JOINER_TIMOUT)
+        self.commissioner.commissioner_add_joiner_with_discerner(DISCERNER2, D_LEN2, PSK2, JOINER_TIMOUT)
+
+        # Add `joiner3` using EUI64 Hardware Address of joiner3.
+        joiner3_hw_addr = self.joiner3.get(wpan.WPAN_HW_ADDRESS)[1:-1]  # Remove the `[]`
+        self.commissioner.commissioner_add_joiner(joiner3_hw_addr, PSK3, JOINER_TIMOUT)
```
Comment (priyankaCh22) [inline silk/tests/openthread/ot_test_meshcop_joiner_discerner.py:135]: For adding joiner3 seeing joiner-add failure.
Please find below details of the error: 
---------
[2020-10-13 22:46:26,452] [silk.Dev-5B97] [DEBUG] sudo ip netns exec ttyACM6 /usr/local/bin/wpanctl -I ttyACM6 commissioner joiner-add F4CE361A35F7EB9E 500 THELAST0FUS
[2020-10-13 22:46:26,503] [silk.Dev-5B97.wpantund] [DEBUG] wpantund[17221]: SpinelNCPTaskSendCommand.cpp:351: Requirement Failed ((mRetVal) == 0)
[2020-10-13 22:46:26,504] [silk.Dev-5B97.wpantund] [DEBUG] wpantund[17221]: SendCommand task encountered an error: 15335435 (0x00EA000B)
[2020-10-13 22:46:26,506] [silk.Dev-5B97] [DEBUG] [stdout] commissioner joiner-add failed with error 15335435. NCP-Specific Errorcode

thanks

File: silk/tests/openthread/ot_test_meshcop_joiner_discerner.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        verify(int(self.joiner3.get(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH), 0) == 0)
+        verify(int(self.joiner3.get(wpan.WPAN_THREAD_JOINER_DISCERNER_VALUE), 0) == 0)
+
+    @testcase.test_method_decorator
+    def test05_add_joiners_on_commissioner(self):
+        self.commissioner.commissioner_start()
+        self.wait_for_completion(self.device_list)
+
+        # Add `joiner1` and `joiner2` using joiner discerner.
+        self.commissioner.commissioner_add_joiner_with_discerner(DISCERNER1, D_LEN1, PSK1, JOINER_TIMOUT)
```
Comment (qilinkang) [inline silk/tests/openthread/ot_test_meshcop_joiner_discerner.py:130]: please check if commissioner_add_joiner_with_discerner() has been added to wpantund_base.py

File: silk/tests/openthread/ot_test_meshcop_joiner_discerner.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+from silk.tools import wpan_table_parser
+from silk.tools.wpan_util import verify, verify_within, is_associated
+from silk.utils import process_cleanup
+
+hwr.global_instance()
+WAIT_TIME = 2  # seconds
+
+PSK1 = 'UNCHARTEDTHEL0STLEGACY'
+PSK2 = 'UNCHARTED4ATH1EFSEND'
+PSK3 = 'THELAST0FUS'
```
Comment (tzm41) [inline silk/tests/openthread/ot_test_meshcop_joiner_discerner.py:31]: Maybe these should be random strings?

File: silk/tests/openthread/ot_test_meshcop_joiner_discerner.py (hunk 2 above)
Comment (priyankaCh22) [reply silk/tests/openthread/ot_test_meshcop_joiner_discerner.py:130]: Sure. Checked this function is present in the master.

**Review by qilinkang — COMMENTED** [review]
Please post the test result.

**Comment by priyankaCh22** [comment]
> Please post the test result.
Sure here it is.
[silk_mesh_discern.log](https://github.com/openthread/silk/files/5449596/silk_mesh_discern.log)

File: silk/tests/openthread/ot_test_meshcop_joiner_discerner.py (hunk 3 above)
Comment (priyankaCh22) [reply silk/tests/openthread/ot_test_meshcop_joiner_discerner.py:31]: Ack.

File: silk/tests/openthread/ot_test_meshcop_joiner_discerner.py (hunk 1 above)
Comment (priyankaCh22) [reply silk/tests/openthread/ot_test_meshcop_joiner_discerner.py:135]: This issue was due to missing flag  OPENTHREAD_CONFIG_COMMISSIONER_MAX_JOINER_ENTRIES in ot build.
Fixed it.

File: silk/tests/openthread/ot_test_meshcop_joiner_discerner.py (hunk 3 above)
Comment (priyankaCh22) [reply silk/tests/openthread/ot_test_meshcop_joiner_discerner.py:31]: Tried random keys but got error. 

e.g. PSK2=random_string(20)
[2020-10-30 23:04:29,618] [silk.Dev-ABDD] [DEBUG] sudo ip netns exec ttyACM10 /usr/local/bin/wpanctl -I ttyACM10 joiner --join FXHHBGNZSMPLRWXRJFDC
[2020-10-30 23:04:29,641] [silk.Dev-ABDD] [DEBUG] [stdout] PSKd "FXHHBGNZSMPLRWXRJFDC" must not contain I, O, Q, or Z characters
[2020-10-30 23:04:29,642] [silk.Dev-ABDD] [DEBUG] [stdout] joiner: error: Invalid PSKd 2

Thus let's keep them as such.

**Comment by priyankaCh22** [comment]
Please find attached PASS results. Thanks
[Uploading mesh_discern_pass.log…]()

**Review by qilinkang — APPROVED** [review]
LGTM

**Review by jwhui — APPROVED** [review]

**Pull request merged**